widget/embedded/src/hooks/useLanguage.ts

```diff
@@ -0,0 +1,15 @@
+import { DEFAULT_LANGUAGE, type Language } from '@rango-dev/ui';
```

Comment:
You can extend the functionality of this hook to be single source of truth for our widget.
I suggest this interface for output of this hook:

```
interface UseLanguage{
  languages: LanguageItem[];
  defaultLanguage: string;
  activeLanguage: string;
  changeLanguage: (language: string) => void;  
}
```

Note 1: 
re-export your `LANGUAGES` (`widget/embedded/src/constants/languages.ts`) as `languages` from this hook. then update `widget/embedded/src/pages/LanguagePage.tsx` to use `languages` from this hook instead of importang from constants directly.

Note 2:
move your useEffect from this file to `widget/embedded/src/Widget.tsx` and use `changeLanguage`.

Reply:
The whole idea is this hook reading, updating and returning information about language.